apps/antalmanac/src/lib/enrollmentHistory.ts

```diff
@@ -0,0 +1,142 @@
+        const queryString = `{
+            enrollmentHistory(department: "${department}", courseNumber: "${courseNumber}", sectionType: Lec) {
+                year
+                quarter
+                department
+                courseNumber
+                dates
+                totalEnrolledHistory
+                maxCapacityHistory
+                waitlistHistory
+                instructors
+            }
+        }`;
+
+        const res =
+            (await queryGraphQL<EnrollmentHistoryGraphQLResponse>(queryString))?.data?.enrollmentHistory ?? null;
+
+        if (res && res.length > 0) {
+            // Before caching and returning the response, we need to do
+            // some parsing so that we can pass the data into the graph
+            const parsedEnrollmentHistory = this.parseEnrollmentHistoryResponse(res);
+
+            // Sort the enrollment history so that the most recent quarters are
+            // in the beginning of the array
+            this.sortEnrollmentHistory(parsedEnrollmentHistory);
+
+            // For now, just return the enrollment history of the most recent quarter
+            // instead of the entire array of enrollment histories
+            const latestEnrollmentHistory = parsedEnrollmentHistory[0];
+            this.enrollmentHistoryCache[cacheKey] = latestEnrollmentHistory;
+            return latestEnrollmentHistory;
+        }
+
+        return null;
+    };
+
+    parseEnrollmentHistoryResponse = (res: EnrollmentHistoryGraphQL[]): EnrollmentHistory[] => {
+        const parsedEnrollmentHistory: EnrollmentHistory[] = [];
+
+        for (const enrollmentHistory of res) {
+            const enrollmentDays: EnrollmentHistoryDay[] = [];
+
+            for (const [i, date] of enrollmentHistory.dates.entries()) {
+                const d = new Date(date);
+                const formattedDate = `${d.getMonth() + 1}/${d.getDate() + 1}/${d.getFullYear()}`;
+
+                enrollmentDays.push({
+                    date: formattedDate,
+                    totalEnrolled: Number(enrollmentHistory.totalEnrolledHistory[i]),
+                    maxCapacity: Number(enrollmentHistory.maxCapacityHistory[i]),
+                    waitlist:
+                        enrollmentHistory.waitlistHistory[i] === 'n/a'
+                            ? null
+                            : Number(enrollmentHistory.waitlistHistory[i]),
+                });
+            }
+
+            parsedEnrollmentHistory.push({
+                year: enrollmentHistory.year,
+                quarter: enrollmentHistory.quarter,
+                department: enrollmentHistory.department,
+                courseNumber: enrollmentHistory.courseNumber,
+                days: enrollmentDays,
+                instructors: enrollmentHistory.instructors,
+            });
+        }
+
+        return parsedEnrollmentHistory;
+    };
+
+    sortEnrollmentHistory = (enrollmentHistory: EnrollmentHistory[]) => {
```

Comment:
Please add a short docstring that clarifies the sorting order instead of relying on the comment on in invokation.